src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -219,6 +232,94 @@ public ObservableList<DeliveryJob> getSortedDeliveryJobList() {
         return FXCollections.observableArrayList(sortedDeliveryJobs);
     }
 
+    @Override
+    public void updateSortedDeliveryJobListByDate() {
+        updateSortedDeliveryJobList(SORTER_BY_DATE);
+
+        for (int i = 0; i < sortedDeliveryJobs.size(); i++) {
+            LocalDate jobDate = sortedDeliveryJobs.get(i).getDeliverDate();
+            String jobSlot = sortedDeliveryJobs.get(i).getDeliverSlot();
+            int slotIndex = Integer.parseInt(jobSlot) - 1;
+            DeliveryJob toAdd = sortedDeliveryJobs.get(i);
+            if (jobListGroupedByDate.containsKey(jobDate)) {
+                ArrayList<ArrayList<DeliveryJob>> jobsInCurrentSlot = jobListGroupedByDate.get(jobDate);
+                if (jobsInCurrentSlot.size() == 0) {
+                    jobsInCurrentSlot = createEmptyDayJobList();
+                }
+                jobsInCurrentSlot.get(slotIndex).add(toAdd);
+                jobListGroupedByDate.put(jobDate, jobsInCurrentSlot);
+            } else {
+                ArrayList<ArrayList<DeliveryJob>> newDateJobList = createEmptyDayJobList();
+                newDateJobList.get(slotIndex).add(toAdd);
+                jobListGroupedByDate.put(jobDate, newDateJobList);
+            }
+        }
+    }
+
+    /**
+     * Update list of week delivery job list,
+     * period spans from 1 week before given date
+     * to 1 week after given date
+     * @param date given/focused date
+     */
+    @Override
+    public void updateWeekDeliveryJobList(LocalDate date) {
+        requireNonNull(date);
+        weekJobListGroupedByDate.clear();
+        this.focusDate = date;
+
+        LocalDate lastWeek = date.minusWeeks(1);
+        LocalDate nextWeek = date.plusWeeks(1);
+        int focusDayOfWeek = date.getDayOfWeek().getValue();
+        int dayOfWeekTracker = 1;
+
+        while (dayOfWeekTracker < 8) {
+            int dayToAdd = dayOfWeekTracker - focusDayOfWeek;
+            LocalDate dayInWeek = date.plusDays(dayToAdd);
+            addWeekJobList(dayInWeek);
+            dayOfWeekTracker++;
+        }
+    }
+
+    @Override
+    public void updateFocusDate(LocalDate jobDate) {
+        this.focusDate = jobDate;
+    }
+    private void addWeekJobList(LocalDate dayToAdd) {
+        if (jobListGroupedByDate.containsKey(dayToAdd)) {
+            ArrayList<ArrayList<DeliveryJob>> currentJobList = jobListGroupedByDate.get(dayToAdd);
+            weekJobListGroupedByDate.put(dayToAdd, currentJobList);
+        } else {
+            weekJobListGroupedByDate.put(dayToAdd, null);
+        }
+    }
+
+    private ArrayList<ArrayList<DeliveryJob>> createEmptyDayJobList() {
+        return new ArrayList<ArrayList<DeliveryJob>>(100);
```

Comment:
I believe arraylist is a dynamic data structure already, so i'm not sure why this is needed....

Reply:
oh haha i left it there cuz wasn't sure how many job slots we r doing so i created a function to make it easier to change ltr